### PR TITLE
Fix custom payload decoding

### PR DESCRIPTION
**Pull request by anlinguist**
This PR fixes the custom payload decoding in embedded messages. Should work on most standard JSON structures. Tested with the following payload:
```
{
  "test": null,
  "testObj": [
    {
      "testObjKey1": 1,
      "testObjKey2": "two"
    },
    {
      "testObj2Key1": {
        "Obj2NestedKey1": 1,
        "obj2NestedKey2": "two"
      },
      "testObj2key2": "2"
    }
  ],
  "messageImportance": "low"
}
```

### CODE REVIEW

File: swift-sdk/Internal/EmbeddedMessagingManager.swift [hunk 1]
```diff
@@ -157,7 +157,7 @@ class EmbeddedMessagingManager: NSObject, IterableEmbeddedMessagingManagerProtoc
             apiClient.track(embeddedMessageReceived: message)
 //            IterableAPI.track(event: "embedded-messaging",
 //                              dataFields: ["name": "received",
-//                                           "messageId": message.metadata.id]
+//                                           "messageId": message.metadata.messageId]
```
Comment (evantk91) [inline swift-sdk/Internal/EmbeddedMessagingManager.swift:160]: Seems like this section needs to be cleaned up on a separate PR. Looks like a function was added in the impression tracking PR.

File: swift-sdk/Internal/EmbeddedMessagingSerialization.swift [hunk 2]
```diff
@@ -4,6 +4,38 @@
...
+                dictionary[key] = val.value
+            }
+            value = dictionary
+        } else {
+            throw DecodingError.dataCorruptedError(in: container, debugDescription: "Container contains an unexpected type")
+        }
+    }
+}
+
+
```
Comment (evantk91) [inline swift-sdk/Internal/EmbeddedMessagingSerialization.swift:38]: This is great! But I thought we were just passing along the custom payload as a JSON. But, if this is the route we want to go, we can do something similar in Android.

File: swift-sdk/IterableEmbeddedMessage.swift [hunk 3]
```diff
@@ -61,5 +62,10 @@ extension IterableEmbeddedMessage {
             public let type: String
             public let data: String?
         }
+        
+        public struct EmbeddedMessageElementsDefaultAction: Codable {
+            public let type: String
+            public let data: String?
+        }
```
Comment (evantk91) [inline swift-sdk/IterableEmbeddedMessage.swift:69]: Nice I forgot about this change. good catch!

**Review by evantk91 — APPROVED** [review]
Looking good. Just added some notes for future development. Kind of a cursory review just making sure everything makes sense. Please confirm works with BE before merging. 👍

File: swift-sdk/Internal/EmbeddedMessagingSerialization.swift [hunk 4]
```diff
@@ -4,6 +4,38 @@
 
 import Foundation
 
+struct AnyDecodable: Decodable {
```
Comment (Ayyanchira) [inline swift-sdk/Internal/EmbeddedMessagingSerialization.swift:7]: Not sure if there should be a generic `AnyDecodable` struct. InApps also supports custom payload but seem to have another approach to it.

**Review by Ayyanchira — COMMENTED** [review]
Left a comment.
